sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ByteArrayContent.java

```diff
@@ -61,6 +61,12 @@ public ByteBuffer toByteBuffer() {
 
     @Override
     public Flux<ByteBuffer> toFluxByteBuffer() {
-        return Flux.defer(() -> Flux.just(ByteBuffer.wrap(toBytes()).asReadOnlyBuffer()));
+        return Flux.defer(() -> Flux.just(toByteBuffer()));
+    }
+
+    @Override
+    public BinaryDataContent copy() {
+        // Content is durable and read-only, return this BinaryDataContent as the copy.
```

Comment:
is this true if `asReadOnlyBuffer` is removed above?

Reply:
Correct, this is only true if the emitted `ByteBuffer`s are read-only